app/controllers/maps_controller.rb

```diff
@@ -26,12 +26,41 @@ def create
     end
   end
 
+  def edit
+    respond_to :js
+
+    map = current_user.maps.find(params[:id])
+    render layout: "modal", locals: { map: map }
+  end
+
+  def update
```

Comment:
There a lot going on here. If you have any ideas for where else to put this or otherwise to clean it up, that would be nice.

We could _always_ broadcast current map and map selector, but it's not always necessary.

Reply:
I was wondering about something like this for co-ordinating the side-effects of CRUD actions https://github.com/collectiveidea/interactor

Reply:
I'm going to give interactor a try to see how it feels.